lua/telescope/previewers/utils.lua

```diff
@@ -51,29 +51,35 @@ end
 
 --- Attach default highlighter which will choose between regex and ts
 utils.highlighter = function(bufnr, ft)
-  if ft and ft ~= '' then
-    if has_ts and ts_parsers.has_parser(ft) then
-      ts_highlight.attach(bufnr, ft)
-    else
-      vim.cmd(':ownsyntax ' .. ft)
-    end
+  if not(utils.ts_highlighter(bufnr, ft)) then
+    utils.regex_highlighter(bufnr, ft)
   end
 end
 
 --- Attach regex highlighter
 utils.regex_highlighter = function(_, ft)
+  local isHighlighted = false
```

Comment:
Lets not introduce a new var. Do `return true` inside the if and `return false` at the end